packages/gatsby-theme-aio/src/components/Frame/index.js

```diff
@@ -53,7 +53,7 @@ const Frame = ({ src, height = 'calc(100vh - var(--spectrum-global-dimension-siz
       // The iframe to which a connection should be made
       iframe: iframe.current,
       // Manually set origin as auto-detection may fail, as the src of the iframe is set later
-      childOrigin: new URL(iframeSrc).origin,
+      childOrigin: isExternalLink(src) ? new URL(src).origin : window.origin,
```

Comment:
What happens with iframeSrc now? is it redundant or is it needed for the iframe src? I'm not sure I understand if childOrigin is expected to be the same as assigned src to Iframe later.

Reply:
And BTW I didn't verify but FYI window will be undefined if executed server-side so make sure it's only executed client-side.

Reply:
> What happens with iframeSrc now

`iframeSrc` is still used to set the actual `src` property of the iframe: https://github.com/adobe/aio-theme/blob/c7fea28efe3c33eba6eb5382e9a67cf99d856522/packages/gatsby-theme-aio/src/components/Frame/index.js#L129

We manually set `iframeOrigin` because Penpal tries to auto-detect this property from the `src` of the iframe (=`iframeSrc`). Auto detection will fail we don't set the `src` of the iframe until the Penpal connection is ready. We need to set the Penpal connection first because Penpal requires that the parent connection is ready before the child tries to connect. If we set the iframe's src property, the child will be loaded and could already try to connect to the parent before the parent connection is ready. This was causing connection failures before https://github.com/adobe/aio-theme/pull/833.

> I'm not sure I understand if childOrigin is expected to be the same as assigned src to Iframe later.

It is not the same, `childOrigin` is the Origin only, while `iframeSrc` is either the full url (if external link) or the path name (if not external).

Reply:
> FYI window will be undefined if executed server-side

Good point, do you know how I could verify that? I think with Gatsby, if you go to a url directly, it should serve the SSG version, right?

Reply:
Ok thanks for clarifying. LGTM